### ISSUE TITLE
Trouble adding new Time and Layouts

### ISSUE THREAD

**Issue by ShreyasAyyengar** [DevLeoko/AdvancedBan#363]
_What version of AdvancedBan (`/AdvancedBan`) are you using?_
 2.1.9
 
_What kind of server do you have (`Bungeecord/single server`)?_
 Bungeecord

_Please provide the EXACT steps required to reproduce the problem..._
1. Install AdvancedBan in your bungeecord proxy

_Any error/log post it through [pastebin.com](http://pastebin.com) and link it here._
None

This isn't really a bug but more of a question.

How do I add my own #TimeLayout and @Layout without editing the ones that are already in the Messages.yml file? I want to create several time and lays outs. Can anyone help?

I have attached a picture. The one that says LightHacking was the example that had come with the plugin. I wanted to create an additional #TimeLayout called #ModerateHacking so that when I run the Command /tempban (user) #ModerateHacking (reason) the time would refer to the one I had created. But when I copy and paste the time layout it came with, after I change a few things the console says that The layout called ModerateHacking wasn’t found. I made sure I’ve spelt everything correctly with the appropriate case but nothing seems to work.

![Config, AdvancedBan](https://user-images.githubusercontent.com/65931801/84591908-99eb5e80-ae74-11ea-93b6-075d446e6bd9.jpeg)

Any solid answers will be greatly appreciated!

**Comment by DevLeoko** [comment]
Remove the "Time:" in the middle. It should be like a folder-structure

**Comment by ShreyasAyyengar** [comment]
Ahh okay. For some reason if I keep a space between the 2 time layouts even with the Time: it works fine. Not sure why it didn’t work last time. 

Since I am here, I just needed your help on one more thing. I have a report plugin that handles all incoming and active reports from players and staff. The GUI enables you to punish the offender once the report is true. The plugin allows you to put whatever command you want when the punishing happens. And because I am using Advancedban I put “tempban _Reported_ #Hacking @Hacking. The problem is, I have quite a few plugins that have the command /ban or tempban and my reports plugin doesn’t know which one to execute. So I think by default it executed the Minecraft ban command. When my advanced ban is installed on my proxy I do not get any plugin prefixes such as /advancedban:tempban 
/advancedban:warn
So I can not specify which plugin I want the ban to happen from. 

My only work around for this was to install advancedban on my backend servers and proxy as well and hook them up with MySQL but it causes other complications. Like if I ban someone using the above setup. Before the prefix “AdvancedBan >> You have been banned” it would show 

**“Kicked whilst connecting to lobby:** AdvancedBan >> you have been banned.”  

The extra “kicked whilst connecting to lobby” doesn’t happen when I only have the plugin in my proxy. 

So basically, it would be really helpful if you could find a way to get the advancedban:(command) prefixes thing to work when it is installed on my proxy. 

Thanks for your great help!

Shreyas

**Comment by DevLeoko** [comment]
Your issue is not as much that AdvancedBan does not have any command prefix. It is that you can not execute bungeecord commands from spigot plugins/console.

You will need to setup BungeeCord <=> Spigot command-based communication. You can do this with a plugin such as ExecuteEverywhere (https://www.spigotmc.org/resources/executeeverywhere.524/) or CommandSyncReborn (https://www.spigotmc.org/resources/commandsync-reborn.63194/).

**Comment by ShreyasAyyengar** [comment]
Ahhh okay!  Thanks for that,

Also, is there anyway to create a custom Punishment ID. Because my anti cheat bans people regularly and it would be nice if I could have a custom punishment ID like “AFH637G99”. 

What’s the difference between Punishment ID and Punishment HEXID

**Comment by DevLeoko** [comment]
There is no such thing as custom Punishment IDs. But why not just use the regular ones? You could add an "AFH" in front of them if that helps :D

HexIDs are the same as IDs just that they are displayed in hexadecimal. So 123456 would be 1E240
https://en.wikipedia.org/wiki/Hexadecimal
